nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ListenSMTP.java

```diff
@@ -0,0 +1,383 @@
+        @WritesAttribute(attribute = "smtp.helo", description = "The value used during HELO"),
+        @WritesAttribute(attribute = "smtp.certificates.*.serial", description = "The serial numbers for each of the " +
+                "certificates used by an TLS peer"),
+        @WritesAttribute(attribute = "smtp.certificates.*.principal", description = "The principal for each of the " +
+                "certificates used by an TLS peer"),
+        @WritesAttribute(attribute = "smtp.from", description = "The value used during MAIL FROM (i.e. envelope)"),
+        @WritesAttribute(attribute = "smtp.to", description = "The value used during RCPT TO (i.e. envelope)")})
+
+public class ListenSMTP extends AbstractProcessor {
+    public static final String SMTP_HELO = "smtp.helo";
+    public static final String SMTP_FROM = "smtp.from";
+    public static final String SMTP_TO = "smtp.to";
+    public static final String MIME_TYPE = "message/rfc822";
+
+
+    protected static final PropertyDescriptor SMTP_PORT = new PropertyDescriptor.Builder()
+            .name("SMTP_PORT")
+            .displayName("Listening Port")
+            .description("The TCP port the ListenSMTP processor will bind to." +
+                    "NOTE that on Unix derivative operating  systems this port must " +
+                    "be higher than 1024 unless NiFi is running as with root user permissions.")
+            .required(true)
+            .expressionLanguageSupported(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    protected static final PropertyDescriptor SMTP_HOSTNAME = new PropertyDescriptor.Builder()
+            .name("SMTP_HOSTNAME")
+            .displayName("SMTP hostname")
+            .description("The hostname to be embedded into the banner displayed when an " +
+                    "SMTP client connects to the processor TCP port .")
+            .required(true)
+            .expressionLanguageSupported(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    protected static final PropertyDescriptor SMTP_MAXIMUM_CONNECTIONS = new PropertyDescriptor.Builder()
+            .name("SMTP_MAXIMUM_CONNECTIONS")
+            .displayName("Maximum number of SMTP connection")
+            .description("The maximum number of simultaneous SMTP connections.")
+            .required(true)
+            .expressionLanguageSupported(false)
+            .addValidator(StandardValidators.INTEGER_VALIDATOR)
+            .build();
+
+    protected static final PropertyDescriptor SMTP_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("SMTP_TIMEOUT")
+            .displayName("SMTP connection timeout")
+            .description("The maximum time to wait for an action of SMTP client.")
+            .required(true)
+            .expressionLanguageSupported(false)
+            .addValidator(StandardValidators.INTEGER_VALIDATOR)
```

Comment:
Using TIME_PERIOD_VALIDATOR would make it clearer what time unit should be used.